addons/mail/static/src/core/common/store_service.js

```diff
@@ -230,6 +211,44 @@ export class Store extends BaseStore {
         return fetchDeferred;
     }
 
+    /**
+     * Create a cacheable version of the `fetchData` method. The result of the
+     * request is cached once acquired. In case of failure, the deferred is
+     * rejected and the cache is reset allowing to retry the request when
+     * calling the function again.
+     *
+     * @param {{[key: string]: boolean}} params Parameters to pass to the `fetchData` method.
+     * @returns {{
+     *      fetch: () => import("@web/core/utils/concurrency").Deferred,
```

Comment:
Wouldn't it possible to define is as a Promise with the result of `fetchData` instead?